### PR TITLE
Document a few Models

### PR DESCRIPTION
**Pull request by Miepee**
Audio, AudioGroups, Background and Object have all been fully documented + been transferred to file scoped namespaces.
Code has most of it documented, but not all, because I'm not familiar enough with the bytecode instructions in order to document them.

### CODE REVIEW

File: DogScepterLib/Core/Models/GMObject.cs [hunk 1]
```diff
@@ -1,196 +1,362 @@
...
+    /// An event that a <see cref="GMObject"/> uses.
+    /// </summary>
+    public class Event : IGMSerializable
+    {
+        /// <summary>
+        /// The subtype of this event.
+        /// </summary>
+        /// <remarks>GameMaker suffixes <see cref="GMCode"/>.<see cref="GMCode.Name"/>, which
+        /// <see cref="Action"/>.<see cref="Action.CodeID"/> references, with this ID.</remarks>
+        //TODO: UMT has this as an UINT, while this here is an INT. Who's correct?
```
Comment (colinator27) [inline DogScepterLib/Core/Models/GMObject.cs:275]: I don't think the number gets high enough to matter, but it's most likely _technically_ a normal int.

File: DogScepterLib/Core/Models/GMObject.cs [hunk 2]
```diff
@@ -1,196 +1,362 @@
...
+        /// <summary>
+        /// An action that an <see cref="Event"/> can execute.
+        /// </summary>
+        public class Action : IGMSerializable
         {
-            public int Subtype;
-            public GMUniquePointerList<Action> Actions;
+            //TODO: a lot of unknown hardcoded values, from pre studio DnD era. Nowadays, DnD blocks are compiled to bytecode.
+            // Some of the attribute names may also be wrong, most of them have been taken from here:
+            // https://github.com/WarlockD/GMdsam/blob/26aefe3e90a7a7a1891cb83f468079546f32b4b7/GMdsam/GameMaker/ChunkTypes.cs#L466
```
Comment (colinator27) [inline DogScepterLib/Core/Models/GMObject.cs:304]: These weren't actually taken from there, these are the real names (you can see them in the 1.x project files)

File: DogScepterLib/Core/Models/GMObject.cs (hunk 1 above)
Comment (colinator27) [reply DogScepterLib/Core/Models/GMObject.cs:275]: I don't think so, but I'm pretty sure the type is still just `int`.

File: DogScepterLib/Core/GMData.cs [hunk 3]
```diff
@@ -78,7 +78,7 @@ public class GMVersionInfo
         public bool RoomObjectPreCreate = false;
 
         /// <summary>
-        /// Whether the GameMaker data file contains various variable counts whose purpose is currently unknown.
+        /// Whether some unknown variables in the GameMaker data file have different values.
```
Comment (colinator27) [inline DogScepterLib/Core/GMData.cs:81]: Oh, this field is referring to the variable counts in one of the chunks (I forget exactly where). There's separate fields for instance/global variables, and this is `false` when those values end up being always the same anyway (on certain GameMaker versions). The counts, I believe, are used to instantiate the correct amount of memory for GML variables, though some may be unused as well.

Comment (Miepee) [reply DogScepterLib/Core/GMData.cs:81]: Oh I forgot to specify the chunk whoops.
They're in "VARI", I'm aware of the usage of this value, but i probably should've clarified them a bit more.

**Pull request merged**